graylog2-server/src/main/java/org/graylog2/shared/metrics/MetricUtils.java

```diff
@@ -187,6 +187,20 @@ public static MetricFilter filterSingleMetric(String name) {
         }
     }
 
+    public static <T extends Metric> T getOrRegister(MetricRegistry metricRegistry, String name, T newMetric) {
```

Comment:
Can you please extend the `MetricUtilsTest` class for this new method? Thank you! :smiley: